client/src/sections/resources/IndustrialFireExtinguisher.tsx

```diff
@@ -0,0 +1,97 @@
+import Line from "../../components/Line";
+import Tile from "../../components/Tile";
+import useHave from "../../hooks/useHave";
+import useGet from "../../hooks/useGet";
+import { Step, useQuestStep } from "../../hooks/useQuest";
+import { $item } from "../../util/makeValue";
+
+const IndustrialFireExtinguisher = () => {
+    const foam = useGet("_fireExtinguisherCharge");
+
+    const knobUnfinished = useQuestStep("questL05Goblin") < Step.FINISHED;
+    const knobUnused = !useGet("fireExtinguisherHaremUsed");
+    const hasPants = useHave($item`Knob Goblin harem pants`);
+    const hasVeil = useHave($item`Knob Goblin harem veil`);
+    const canVisitTheThrone = hasPants && hasVeil;
+    const showKnob =
+        useQuestStep("questL05Goblin") > Step.UNSTARTED &&
+        !canVisitTheThrone &&
+        knobUnused &&
+        knobUnfinished;
+
+    const bridgeUnfinished = useQuestStep("questL09Topping") < 1;
+    const blechUnused = !useGet("fireExtinguisherChasmUsed");
+    const showBlech =
+        useQuestStep("questL09Topping") > Step.UNSTARTED &&
+        bridgeUnfinished &&
+        blechUnused;
+
+    const batwallsUnfinished = useQuestStep("questL04Bat") < 3;
+    const batUnused = !useGet("fireExtinguisherBatHoleUsed");
+    const showBat =
+        useQuestStep("questL04Bat") > Step.UNSTARTED &&
+        batwallsUnfinished &&
+        batUnused;
+
+    const bonerUnready = [
```

Comment:
i still don't like this weird map and general structure and i finally figured out one of the reasons why -- while bonerUnready works, despite going to such extremes to not declare variables for the separate evilness, you actually don't ever check in showCyrpt to see if any zone has 10 evilness left, because you really should not waste a charge if you have <10 left in all distinct zones. so you did a lot of work and still have to do something else with all those useGets lol. the structure can be kept but please add that somehow.